src/parser.ts

```diff
@@ -2412,13 +2427,13 @@ function parseVariableDeclaration(options) {
     const id = parsePattern(params, 'var');
 
     // ECMA-262 12.2.1
-    if (scanner.strict && scanner.isRestrictedWord(id.name)) {
+    if (state.strict && scanner.isRestrictedWord(id.name)) {
```

Comment:
In the subsequent improvement, I play with that idea. Remains to be seen whether it is worth it since it won't be called too many times anymore.